src/Future.js

```diff
@@ -10,6 +10,16 @@ function jail(handler, f){
   };
 }
 
+function fork(m, reject, resolve) {
```

Comment:
I think for that particular case my suggestion would be to use `Future(sinon.spy())` instead.

I'm personally in favour of sticking with just the first commit in this PR, but I welcome input from others. /cc @buzzdecafe @TheLudd